### DISCUSSION TITLE
add_basemap shows unusual behaviour and does not throw error on misspelling layername

### DISCUSSION THREAD

**Discussion by amanbagrecha** [opengeos/leafmap#5] [General]
I was trying out [this](https://leafmap.gishub.org/notebooks/02_using_basemaps/) example and tried adding a basemap which is not in the `ee_basemaps` list as shown in the image. To my surprise it did not throw any error.
![](https://i.imgur.com/GXWmamU.png)

Similar behavior on trying `leaflet.Map(google_map="HYbRIDss")`. It does not throw any error if the argument is not in the list.

Additionally, even if the argument is correct and the layer is already in `self.layers` it stills adds another layer on top of the existing one as shown in the image. 
![](https://i.imgur.com/7gp2Rcv.jpg)
I think the problem stems from [this](https://github.com/giswqs/leafmap/blob/master/leafmap/leafmap.py#L178) line. The `if` statement returns `false` instead of throwing an error.

Was not sure if this is a bug so added in discussion. Please guide if there is something I am doing wrong. Thank you!

**Comment by giswqs** [comment]
@amanbagrecha Thanks for reporting. This bug has been fixed (https://github.com/giswqs/leafmap/commit/44a253014aaf8fa0d083bc9c904875897213965e). Please update the package using `leafmap.update_package()` and try again.

**Comment by berbagi12** [comment]
good sir
